modules/iyzicocheckoutform/iyzicocheckoutform.php

```diff
@@ -148,8 +142,97 @@ protected function _postProcess()
+        $this->context->smarty->assign('link', $test);
+        if ($version == $response['iyzico_version']) {
+            if (isset($_GET['updated_iyzico'])) {
+
+                $version_updatable = $_GET['updated_iyzico'];
+
+                function recurse_copy($src, $dst) {
+                    $dir = opendir($src);
+                    @mkdir($dst);
+                    while (false !== ( $file = readdir($dir))) {
+                        if (( $file != '.' ) && ( $file != '..' )) {
+                            if (is_dir($src . '/' . $file)) {
+                                recurse_copy($src . '/' . $file, $dst . '/' . $file);
+                            } else {
+                                copy($src . '/' . $file, $dst . '/' . $file);
+                            }
+                        }
+                    }
+                    closedir($dir);
+                }
+
+                function rrmdir($dir) {
+                    if (is_dir($dir)) {
+                        $objects = scandir($dir);
+                        foreach ($objects as $object) {
+                            if ($object != "." && $object != "..") {
+                                if (filetype($dir . "/" . $object) == "dir")
+                                    rrmdir($dir . "/" . $object);
+                                else
+                                    unlink($dir . "/" . $object);
+                            }
+                        }
+                        reset($objects);
+                        rmdir($dir);
+                    }
+                }
+
+                $ch = curl_init();
+                curl_setopt($ch, CURLOPT_URL, 'http://iyzico.kahvedigital.com/update');
+                curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+                curl_setopt($ch, CURLOPT_POSTFIELDS, "new_version=$version_updatable");
+                $response = curl_exec($ch);
+                $response = json_decode($response, true);
+                $serveryol = $_SERVER['DOCUMENT_ROOT'];
+                $ch = curl_init();
+                $source = $response['file_dest'];
+                curl_setopt($ch, CURLOPT_URL, $source);
+                curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+                $data = curl_exec($ch);
+                curl_close($ch);
+                $foldername = $response['version_name'];
+                $fullfoldername = $serveryol . '/' . $foldername;
+                if (!file_exists($fullfoldername)) {
+                    mkdir($fullfoldername);
+                }
+                if (file_exists($fullfoldername)) {
+                    $unzipfilename = 'iyzicoupdated.zip';
+                    $file = fopen($fullfoldername . '/' . $unzipfilename, "w+");
+                    fputs($file, $data);
+                    fclose($file);
+                    $path = pathinfo(realpath($fullfoldername . '/' . $unzipfilename), PATHINFO_DIRNAME);
+                    $zip = new ZipArchive;
```

Comment:
$res değişkeni verdiğin zip dosyasının açılıp açılamadığının sonucunu döndürüyor. ZipArchive class'ı ile ilgili bir kontrol değil. Eğer ZipArchive class'ı mevcut değilse burada fatal yiyeceğini öngörüyorum. Düzeltmemiz lazım.